test/server/cards/01_SOR/units/BobaFettDisintegrator.spec.js

```diff
@@ -0,0 +1,115 @@
+                this.player1.clickCard(this.lomPyke);
+
+                // check board state
+                expect(this.player2).toBeActivePlayer();
+                expect(this.lomPyke.damage).toBe(3);
+                expect(this.bobaFett.damage).toBe(4);
+
+                // reset state
+                this.bobaFett.damage = 0;
+                this.bobaFett.exhausted = false;
+                this.lomPyke.damage = 0;
+
+                // Case 3 Ability activates when attacking an exhausted unit not played in this phase
+                this.consularSecurityForce.exhausted = true;
+                this.player2.passAction();
+                this.player1.clickCard(this.bobaFett);
+                this.player1.clickCard(this.consularSecurityForce);
+
+                // check board state
+                expect(this.bobaFett.damage).toBe(3);
+                expect(this.consularSecurityForce.damage).toBe(6);
+
+                // reset state
+                this.bobaFett.damage = 0;
+                this.bobaFett.exhausted = false;
+                this.consularSecurityForce.damage = 0;
+                this.consularSecurityForce.exhausted = false;
+                this.player2.passAction();
+
+                // Case 4 Ability should activate when attacking leader deployed previous phase
+                // TODO The card enteres play event doesn't handle leader deployment correctly so we need to wait for the fix before uncommenting this test.
+                /* this.lukeSkywalker.exhausted = true;
+                this.player1.clickCard(this.bobaFett);
+                this.player1.clickCard(this.lukeSkywalker);
+
+                // check board state
+                expect(this.lukeSkywalker.damage).toBe(6);
+                expect(this.bobaFett.damage).toBe(4);*/
+
+                // Case 5 Ability shouldn't activate when selecting BobaFett's ability first.
+                this.player1.clickCard(this.vambraceGrappleshot);
+                this.player1.clickCard(this.bobaFett);
+                this.player2.passAction();
+
+                this.player1.clickCard(this.bobaFett);
+                this.player1.clickCard(this.consularSecurityForce);
+                this.player1.clickPrompt('If this unit is attacking an exhausted unit that didn\'t enter play this round, deal 3 damage to the defender.');
+
+                // check game state
+                expect(this.consularSecurityForce.damage).toBe(5);
+                expect(this.consularSecurityForce.exhausted).toBe(true);
+
+                // reset state
+                this.consularSecurityForce.damage = 0;
+                this.bobaFett.exhausted = false;
+                this.bobaFett.damage = 0;
+                this.consularSecurityForce.exhausted = false;
+
+                // Case 6 Ability should activate when selecting vambrace-grappleshots ability before Boba Fetts.
+                this.player2.clickCard(this.entrenched);
```

Comment:
Why is entrenched needed here? Could we just reset consular security force's damage to 0?

Reply:
The reason why I used entrenched here is to see how much damage consularSecurityForce will receive. I wanted to see if the ability worked with vambrace-grappleshot, and since vambrace gives +2 power, I wouldn't be able to see if the damage is correct or not accurately.